### ISSUE TITLE
squid: Log date/time formatting into gui

### ISSUE THREAD

**Issue by monster900** [opnsense/core#1831]
Hi,
ist it possible to show the squid logs for access an store logs in a more user friendly time format into the gui?
Right now both logs are shown into unix time format.

best regards
Dirk

**Comment by fichtner** [comment]
Hi Dirk,

It's not impossible, but there are far superior tools for Squid Log Analysis than we could ever write / provide. Have you checked out lightsquid?


Cheers,
Franco

**Comment by Stephanowicz** [comment]
Hi,

I've thinking bout this, too - and according to [https://robinward.com/?p=372](https://robinward.com/?p=372) it's quite easy by adding something like
`logformat squid %tl.%03tu %6tr %>a %Ss/%03Hs %`
to the squid config.

This means one needs to add the changes to the proxy template in 

> /usr/local/opnsense/service/templates/OPNsense/Proxy/squid.conf 

I added it to line 337 - this is used when enabling "file extended" in the general settings
**%tl.%03tu %6tr** is added after `logformat opnsense      ` : 
`logformat opnsense      %tl.%03tu %6tr %>a %[ui %>eui %[un [%tl] "%rm %ru HTTP/%rv" %>Hs %<st "%{Referer}>h" "%{User-Agent}>h" %Ss:%Sh
`

After this modification the logfile enties changed from
`1506423311.903     94 192.168.200....`
to
`26/Sep/2017:12:58:12 +0200.969 274530 192.168.200....`

Cheers,

Stephan

**Comment by monster900** [comment]
Stephan,
it's working! But only for the access log. The store log keep showing the unix time format. Is there another line into squid.conf to change the time format for store log too?
What happens after an update? I think that these changes gone after an update, right?

best regards
Dirk

**Comment by fichtner** [comment]
Yes, it should be added to OPNsense else it's overwritten. I don't mind changing the format, but what happens to e.g. lightsquid analysis in these cases?

**Comment by Stephanowicz** [comment]
@fichtner do these depend on the unix time format?

@monster900 as far as I know it's not possible for the store log

probably a custom format field would be helpful... though one needs  to know what one is doing...

**Comment by fichtner** [comment]
I don't know, that's all I'm saying.

lightsquid is not a default so we can get away with using the new timestamp format as a default, but then we need to provide a fallback option for the people who use other tools that expect the "standard" log format.

**Comment by Stephanowicz** [comment]
A combobox with different formatting options would be nice^^

**Comment by fichtner** [comment]
I'm giving this back to the pool. Started on this several times, but I can't find a nice solution.

**Comment by nhirokinet** [comment]
Hi,

I tried to do something about it, and first trial is #3326; this would do something better, but not a clean one.

At least:

- As long as log format are unchangable (like other format of "Mar 14 01:37:13", where number of spaces is hardcoded as `$logsplit`), unix datetime format can be parsed, but this is: "as long as log format are unchangable".
- This seems to be a chance (for me) to change format to more localized, like "2019/03/14 01:37:13" rather than "Mar 14 01:37:13" in Japanese or Chinese setting, but I found hard to parse existing log format in other places, because they lacks year (please see pull request for this).

This, including pull request above, is just a comment, but if some more comments come, I might do something for this (and hopefully asian-friendly date format to other places :) ).
